gandi/resource_livedns_record.go

```diff
@@ -74,29 +81,80 @@ func expandRecordID(id string) (zone, name, recordType string, err error) {
 	return
 }
 
+func keepUniqueRecords(recordsList []string) []string {
+	keys := make(map[string]bool)
+	uniqueRecords := []string{}
+	for _, entry := range recordsList {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			uniqueRecords = append(uniqueRecords, entry)
+		}
+	}
+	return uniqueRecords
+}
+
+func isRecordWrappedWithQuotes(record string) bool {
+	return strings.HasPrefix(record, "\"") && strings.HasSuffix(record, "\"")
+}
+
+func containsRecord(recordsList []string, recordToFind string) int {
```

Comment:
Could you return a tuple such as `(index int, exist bool)` instead of encoding a non existing record with `-1`?